### ISSUE TITLE
Support managing snapshot policies

### ISSUE THREAD

**Issue by swoehrl-mw** [opensearch-project/opensearch-k8s-operator#806]
Discussed in https://github.com/opensearch-project/opensearch-k8s-operator/discussions/796

The operator already supports managing snapshot repositories. A step further would be to add a custom resource and reconciler logic (similar to ISMPolicies) to also manage snapshot policies via the operator.

**Comment by swoehrl-mw** [comment]
@prudhvigodithi 

> so the idea is to have a CRD to configure a snapshot policy ?

Exactly. We can introduce a new CRD for that and add a reconciler to the operator to handle it.

**Comment by cthtrifork** [comment]
Wouldn't creating a ISMPolcy with a snapshot: https://opensearch.org/docs/latest/im-plugin/ism/policies/#snapshot archieve the same?

**Comment by AdaptiveStep** [comment]
Hmm. What opensearch needs is a way to document the api in a systemic way using openAPI/swagger style. That would simplify the usage of the opensearch api a lot for future operator developement. (That is atleast the path elastic took)

opensearch needs an api like this: https://www.elastic.co/docs/api/doc/kibana/v8

This would also make sure that the api- documentation is automatically generated and also up-to-date. I can't believe we don't have something like this yet..

**Comment by rootxrishabh** [comment]
@cthtrifork, ISM acts on state rotation, if an index enters another state then it allows an action such as creating a snapshot, but this does not do periodic snapshots but only once when the index reaches the state. Maybe this is possible with ISM but that would require periodic state change based on a cron-job rather than condition.

**Comment by rootxrishabh** [comment]
@swoehrl-mw @prudhvigodithi, just wanted to mention that I am currently working on building the CRD for snapshot policies.
